agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -440,6 +440,37 @@ impl DeriveSchema for TaggedOperator {
                     })
             }};
         }
+        macro_rules! array_schema_or_error {
```

Comment:
this is used for `$reduce` and `$map`, which tolerate nullish arguments. This allows for null, missing, or valid anyofs to be included.